apollo-router/src/spec/schema.rs

```diff
@@ -20,30 +20,23 @@ use crate::query_planner::OperationKind;
 use crate::Configuration;
 
 /// A GraphQL schema.
-#[derive(Debug)]
 pub(crate) struct Schema {
     pub(crate) raw_sdl: Arc<String>,
-    pub(crate) definitions: Valid<apollo_compiler::Schema>,
+    pub(crate) federation_supergraph: apollo_federation::Supergraph,
```

Comment:
so this is not just an experimental integration that can be switched on without affecting the rest, it directly modifies core code?

Reply:
Yes, as the PR is now fed-next metadata is extracted from the supergraph unconditionally. The default config makes it not used (except for the apollo-compiler schema it contains) but still created.

If we judge the risk of spurious failure in fed-next initialization is too great, this field could be turned into an enum that contains either a `Schema` or a `Supergraph`

Reply:
hmm, we do use the `@link()` handling in API schema, so basic use cases should be okay. I'm not sure if we are 100% confident in it though.

it would be really nice to have a supergraph available because there are a few places that mimick `@link` functionality in the router

Reply:
we have some supergraphs in https://github.com/apollographql/router/tree/dev/examples/graphql